### PR TITLE
DHFPROD-2080: Latest rebased selectors for develop

### PR DESCRIPTION
**Pull request by clockworked247**

### CODE REVIEW

**Comment by ayuwono** [comment]
The same thing on ingest step input file type and output file type options

**Comment by clockworked247** [comment]
Good catch.  I'll fix that.  Thanks.

**Comment by clockworked247** [comment]
Fixed.  If you add a dynamic class name to a select option it does not automatically add the option class that gives it the correct padding.  So, I changed those selectors to IDs instead and it's working great.  

Also updated to match these changes:
https://wiki.marklogic.com/display/ENGINEERING/QuickStart+UI+Selectors+for+Testing

File: web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html [hunk 1]
```diff
@@ -36,14 +36,14 @@ <h1 mat-dialog-title>{{getDialogTitle()}}</h1>
...
-          <input [class]="'source-weights-source-' + i" matInput formControlName="source"/>
+          <input [class]="'source-weights-name-' + i" matInput formControlName="source"/>
         </mat-form-field>
         <mat-icon></mat-icon>
         <mat-form-field>
           <mat-label>Weight</mat-label>
           <input [class]="'source-weights-weight-' + i" matInput formControlName="weight"/>
         </mat-form-field>
-        <button class="remove-source-weight-btn" mat-icon-button (click)="onRemoveSourceWeight(i)">
+        <button [class]="'remove-source-weight-btn-' + i" mat-icon-button (click)="onRemoveSourceWeight(i)">
```
Comment (ayuwono) [inline web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html]: I think we should remove the square brackets on the class? should be:
<button class="'remove-source-weight-btn-' + i" mat-icon-button (click)="onRemoveSourceWeight(i)">

This cause a css/styling bug on the minus button on merge strategy dialog, the minus button has a square background:
Screen Shot 2019-04-11 at 9 51 48 AM

It should look like:
Screen Shot 2019-04-11 at 9 52 08 AM

And the class name should be: remove-merge-strategy-source-weight-btn, just to be consistent with the add-merge-strategy-source-weight-btn

Comment (clockworked247) [reply web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html]: @ayuwono Fixed and updated wiki.  Using IDs now for the source weight buttons for both options and strategies.

**Comment by clockworked247** [comment]
Build passes locally.  Need review approvals from @ayuwono @wooldridge @alexsmr plz!

**Comment by ayuwono** [comment]
> Build passes locally. Need review approvals from @ayuwono @wooldridge @alexsmr plz!

testing this

File: web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html [hunk 2]
```diff
@@ -43,7 +43,7 @@ <h1 mat-dialog-title>{{getDialogTitle()}}</h1>
           <mat-label>Weight</mat-label>
           <input [class]="'source-weights-weight-' + i" matInput formControlName="weight"/>
         </mat-form-field>
-        <button [class]="'remove-source-weight-btn-' + i" mat-icon-button (click)="onRemoveSourceWeight(i)">
+        <button [id]="'remove-merge-option-source-weight-btn-' + i" mat-icon-button (click)="onRemoveSourceWeight(i)">
```
Comment (ayuwono) [inline web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html]: Shouldn't be merge-option

Should be:
<button [id]="'remove-merge-strategy-source-weight-btn-' + i" mat-icon-button (click)="onRemoveSourceWeight(i)">

**Comment by ayuwono** [comment]
@clockworked247 , need one more fix and I think we're good to go

File: web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html (hunk 2 above)
Comment (clockworked247) [reply web/src/main/ui/app/components/flows-new/edit-flow/mastering/merging/ui/add-merge-strategy-dialog.component.html]: Yikes, that was dumb... fixed.

**Comment by clockworked247** [comment]
@wooldridge good catch.  Thanks.  I also searched through the code for any further additions of class name to [class] relating to dropdowns and there are no more.  

Should be good to go...?

**Pull request merged**